Implemented django-tables2 to lobby list view

jiyeongchloe opened the pull request:
To implement django-tables2 to lobby list view, I
- added `django_tables2` to requirements/base.txt
- changed the lobby-list definitions and classes in `urls.py` and `views.py`
- added `tables.py`
- modified `lobby-list.html` to render the table instead of listing the lobbies in bullets

jiyeongchloe committed:
added django-tables2 to lobby list view

jiyeongchloe committed:
trying to fix  issue

nugerhard reviewed (approved):
Works on my end. Maybe some thought needs to be given to what columns are eventually displayed by this lobbies table, but I don't think that's super necessary to figure out right now. Thanks for adding this!

elizabethli31 commented on requirements/base.txt:
```diff
@@ -7,3 +7,4 @@ django-environ==0.10.0  # https://github.com/joke2k/django-environ
 django-allauth==0.57.0  # https://github.com/pennersr/django-allauth
 django-crispy-forms==2.0  # https://github.com/django-crispy-forms/django-crispy-forms
 crispy-bootstrap5==0.7  # https://github.com/django-crispy-forms/crispy-bootstrap5
+django-tables2
```
Remove this - you only need it in local.txt

jiyeongchloe replied:
has been removed

elizabethli31 reviewed (changes requested):
Could you guys update with dev and then I'll review? Because of some changes in the game model in dev, I can't make game objects anymore with the version you guys currently have. Looks fine generally!

jiyeongchloe committed:
Merge branch 'dev' into matches/lobby-django-tables

jiyeongchloe committed:
incorporated comment on PR #158

jiyeongchloe committed:
fixing unused import issue

elizabethli31 reviewed (approved):
LGTM! This view looks really good and is much cleaner than before. Great work Chloe!

jiyeongchloe commented:
**Closing Comment:**
PR is merged and closed.

Chloe: added django_tables2 to requirements/base.txt, changed the lobby-list definitions and classes in urls.py and views.py, added tables.py, modified lobby-list.html to render the table instead of listing the lobbies in bullets
Lucien: helped debug issues and migrations problems
Nat: reviewer

elizabethli31 commented:
Issue Score: Excellent

Comments:
Nice improvement from the rudimentary list view that was previously implemented. 
<!-- {"score": "excellent", "minor": false} -->
